From 98b36c1403860bd70b138aa79ed2317e9b0df223 Mon Sep 17 00:00:00 2001
From: Alan Pevec <alan.pevec@redhat.com>
Date: Fri, 8 Apr 2016 14:59:43 +0200
Subject: [PATCH] Workaround for /dev/kvm permissions issues

Taken from Fedora qemu.spec:
  http://pkgs.fedoraproject.org/cgit/rpms/qemu.git/commit/?id=41aca9586f9f2a573b6b657b236cad38e316c0e8
See https://bugzilla.redhat.com/show_bug.cgi?id=950436

Also remove load_kvm which was EL6 relict:
  https://bugzilla.redhat.com/show_bug.cgi?id=963198

Change-Id: Ia37d19a0397e3853f5a6913e3aa85daf6e3aefa2
---
 packstack/puppet/templates/nova_compute_libvirt.pp | 15 +++++++--------
 1 file changed, 7 insertions(+), 8 deletions(-)

diff --git a/packstack/puppet/templates/nova_compute_libvirt.pp b/packstack/puppet/templates/nova_compute_libvirt.pp
index 3eb6770..c1cce29 100644
--- a/packstack/puppet/templates/nova_compute_libvirt.pp
+++ b/packstack/puppet/templates/nova_compute_libvirt.pp
@@ -16,6 +16,13 @@ exec { 'qemu-kvm':
   command => 'yum install -y -d 0 -e 0 qemu-kvm',
   onlyif  => 'yum install -y -d 0 -e 0 qemu-kvm-rhev &> /dev/null && exit 1 || exit 0',
   before  => Class['nova::compute::libvirt'],
+} ->
+# chmod is workaround for https://bugzilla.redhat.com/show_bug.cgi?id=950436
+file { '/dev/kvm':
+  owner  => 'root',
+  group  => 'kvm',
+  mode   => '666',
+  before => Class['nova::compute::libvirt'],
 }
 
 $libvirt_vnc_bind_host = hiera('CONFIG_IP_VERSION') ? {
@@ -32,14 +39,6 @@ class { '::nova::compute::libvirt':
   libvirt_inject_partition => '-1',
 }
 
-exec { 'load_kvm':
-  user    => 'root',
-  command => '/bin/sh /etc/sysconfig/modules/kvm.modules',
-  onlyif  => '/usr/bin/test -e /etc/sysconfig/modules/kvm.modules',
-}
-
-Class['nova::compute'] -> Exec['load_kvm']
-
 file_line { 'libvirt-guests':
   path    => '/etc/sysconfig/libvirt-guests',
   line    => 'ON_BOOT=ignore',
